ionic_user_interface/src/components/GymRouteList.vue

```diff
@@ -0,0 +1,486 @@
+          :value="{ lower: 0, upper: 14 }"
+          dual-knobs
+          @ionChange="filterGradeHandler"
+          debounce="100"
+        ></ion-range>
+        <div class="sliderticks">
+          <p v-for="index of range(0, 14)" :key="index">V{{ index }}</p>
+        </div>
+      </div>
+      <ion-button
+        expand="full"
+        color="light"
+        slot="end"
+        class="ion-no-margin"
+        @click="setPopoverOpen(true, $event)"
+      >
+        <ion-label>Sort</ion-label>
+        <ion-icon :icon="filterOutline"></ion-icon>
+      </ion-button>
+    </ion-item>
+    <ion-popover
+      :is-open="isPopoverOpen"
+      :translucent="true"
+      :event="popoverEvent"
+      keyboard-close
+      @didDismiss="setPopoverOpen(false)"
+    >
+      <ion-list class="no-padding no-margin">
+        <ion-item button @click="setSortMode(SortMode.NEWEST)">
+          Newest
+          <ion-icon
+            slot="end"
+            :icon="checkmarkOutline"
+            v-if="sortMode === SortMode.NEWEST"
+          ></ion-icon>
+        </ion-item>
+        <ion-item button @click="setSortMode(SortMode.OLDEST)">
+          Oldest
+          <ion-icon
+            slot="end"
+            :icon="checkmarkOutline"
+            v-if="sortMode === SortMode.OLDEST"
+          ></ion-icon>
+        </ion-item>
+        <ion-item button @click="setSortMode(SortMode.VOTES)">
+          Most votes
+          <ion-icon
+            slot="end"
+            :icon="checkmarkOutline"
+            v-if="sortMode === SortMode.VOTES"
+          ></ion-icon>
+        </ion-item>
+        <ion-item button @click="setSortMode(SortMode.EASIEST)">
+          Easiest
+          <ion-icon
+            slot="end"
+            :icon="checkmarkOutline"
+            v-if="sortMode === SortMode.EASIEST"
+          ></ion-icon>
+        </ion-item>
+        <ion-item button @click="setSortMode(SortMode.HARDEST)">
+          Hardest
+          <ion-icon
+            slot="end"
+            :icon="checkmarkOutline"
+            v-if="sortMode === SortMode.HARDEST"
+          ></ion-icon>
+        </ion-item>
+      </ion-list>
+    </ion-popover>
+    <ion-list class="ion-no-margin ion-no-padding">
+      <ion-card
+        v-for="(route, index) of filteredSortedRoutes"
+        :key="index"
+        class="ion-text-left route-card"
+        @click="() => handleRouteCardClick(route.username, route.createdAt)"
+      >
+        <ion-card-header>
+          <ion-card-title>{{ route.routeName }}</ion-card-title>
+          <VoteButton
+            class="vote-button"
+            :username="route.username"
+            :createdAt="route.createdAt"
+            v-model:voteCount="route.voteCount"
+            v-model:hasVoted="route.hasVoted"
+          ></VoteButton>
+        </ion-card-header>
+        <ion-card-content>
+          <b>Creator:&nbsp;</b>
+          <router-link :to="'/userRoutes/' + route.username">{{ route.username }}</router-link>
```

Comment:
When i click here, it goes to `/viewRoute/username/createdAt` instead..

Reply:
fixed